scalalib/src/mill/scalalib/CrossScalaModule.scala

```diff
@@ -33,12 +33,13 @@ object CrossScalaModule {
    * Used with a [[mill.define.Cross]] when you want [[CrossScalaModule]]'s
    * nested within it
    */
-  trait Base extends mill.Module {
-    def crossScalaVersion: String
+  trait Base extends mill.Cross.Module[String] {
+    def crossScalaVersion: String = crossValue
+    def crossPathSegments = List(crossScalaVersion)
     private def wrapperSegments0 = millModuleSegments.parts
     trait CrossScalaModule extends mill.scalalib.CrossScalaModule {
       override def wrapperSegments = wrapperSegments0
-      def crossScalaVersion = Base.this.crossScalaVersion
+      def crossValue = Base.this.crossScalaVersion
```

Comment:
Isn't `crossValue` already defined in inherited Cross.Module (via CrossScalaModule)?

Reply:
Yes, but this is a bit of an unusual case: the actual `Cross.Module` that gets passed to `Cross` is `CrossScalaModule.Base`. These `CrossScalaModules` nested within it are just proxies that share the `crossValue` of the `Base` trait, so we don't need to repeat the `extends Cross[...](...)` for every single submodule

So for these inner `CrossScalaModule`s, the call to `crossValue` just forwards to the `crossValue` of the enclosing `CrossScalaModule.Base`

We could also split up `CrossScalaModule` so there's a `CrossScalaModule0` that foes not have `Cross.Module` mixed in, and then do the forwarding in `def crossScalaVersion` instead. I don't have a strong opinion either way

Reply:
Huh, hard to grasp. I guess understanding this is all about getting how `CrossScalaModule.Base` is supposed to work/used, which isn't obvious from the code itself. We should add some doc to `CrossScalaModule.Base`, describing the concept. An example (or link to it) says probably more than many words.

Reply:
Yeah I'm not totally happy with it, but it's in the examples which I hope to include in the docs. 

In general I'd hope this kind of "base"/"nested" thing to be generic and not tied to `ScalaModule`, but for now this trades off some simplicity for some much needed convenience to defining cross-built Scala projects

Reply:
@lefou I think I've come to a reasonable solution here: 

1. The `CrossScalaModule.Base`/`CrossScalaModule` pairing has been replaced by a more general `Cross.ModuleN`/`InnerCrossModuleN` pairing. Thus any `Cross.ModuleN` can contain `InnerCrossModuleN`s that share the same set of `crossValueN`s

2. Removed the `Cross.Arg[T]` traits, leaving just `Cross.ModuleN`, to allow the `InnerCrossModuleN`s to work properly

Essentially, now "propagate cross values to child modules" is a generic capability that all `Cross.Module`s have, not just `CrossScalaModule`. It's marginally more verbose at the use site (`extends CrossScalaModule with InnerCrossModule` rather than just `extends CrossScalaModule`) but overall should be a reasonable compromise between conciseness, simplicity, and genericity.